src/commands/moderation.ts

```diff
@@ -216,34 +216,23 @@ export class Sticky extends StickyCommand {
 		}
 
 		// Validate @role is not already sticky (database query)
-		this.worker = await pgPool.connect();
-		const res = await this.worker.query('SELECT sticky FROM servers WHERE serverid = $1', [this.guild.id]);
-		if (!res.rows.length) {
+		const res = await database.queryWithResults('SELECT sticky FROM servers WHERE serverid = $1', [this.guild.id]);
+		if (!res.length) {
 			throw new Error(`Unable to find sticky roles in database for guild: ${this.guild.name} (${this.guild.id})`);
 		}
-		const stickyRoles: string[] = res.rows[0].sticky;
+		const stickyRoles: string[] = res[0].sticky;
 
 		if (stickyRoles.includes(role.id)) {
-			this.releaseWorker();
 			return this.errorReply('That role is already sticky!');
 		}
 
 		// ---VALIDATION LINE---
 		// Make @role sticky (database update)
 		stickyRoles.push(role.id);
-		try {
-			await this.worker.query('BEGIN');
-			await this.worker.query('UPDATE servers SET sticky = $1 WHERE serverid = $2', [stickyRoles, this.guild.id]);
-			// Find all users with @role and perform database update so role is now sticky for them
-			await this.massStickyUpdate(role);
-			await this.worker.query('COMMIT');
-			this.releaseWorker();
-		} catch (e) {
-			await this.worker.query('ROLLBACK');
-			this.releaseWorker();
-			throw e;
-		}
-		// Return success message
+		await database.withinTransaction([
+			{statement: 'UPDATE servers SET sticky = $1 WHERE serverid = $2', args: [stickyRoles, this.guild.id]},
+		]);
+		await this.massStickyUpdate(role); // I hope this works...
```

Comment:
Maybe it would be best to change `massStickyUpdate` to return a `Query` instead so we can include it in the above transaction.

Reply:
hmm, sure